tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -694,16 +630,17 @@ bool TRTEngineOp::ExecuteTrtEngine(OpKernelContext* ctx,
   }
 
   const bool kRetry = true;
-  if (trt_context_idx >= 1) {
+  if (trt_context_idx >= engine_context->execution_context.size()) {
     LOG(ERROR) << "Requested engine context with index " << trt_context_idx
-               << ", but only 1 context is present.";
+               << ", but only " << engine_context->execution_context.size()
+               << "contexts are present.";
     return kRetry;
   }
   const int num_binding = cuda_engine->getNbBindings();
   std::vector<void*> buffers(num_binding);
 
   mutex_lock lock(engine_context->mu);
-  auto& execution_context = engine_context->execution_context;
+  auto& execution_context = engine_context->execution_context[trt_context_idx];
```

Comment:
engine_context->execution_context is guarded by a mutex, so need to move the lock to before accessing engine_context->execution_context.size()